components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/util/ParUtil.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.par.util;
+
+import org.wso2.carbon.identity.oauth.par.core.ParAuthService;
+import org.wso2.carbon.identity.oauth.par.core.ParAuthServiceImpl;
+
+/**
+ * Util class for PAR.
+ */
+public class ParUtil {
+
+    private ParUtil() {
+        // private constructor
+    }
+
+    private static ParAuthService parAuthService = new ParAuthServiceImpl();
+
+    /**
+     * Get instance of parAuthService.
+     *
+     * @return Instance of parAuthService.
+     */
+    public static ParAuthService getParAuthService() {
```

Comment:
Instead of having a util to get the ParAuthService lets have a dataholder and add the ParAuthService instance created when registering it as a osgi service and use that same instance through the dataholder.